packages/at_client/lib/src/decryption_service/self_key_decryption.dart

```diff
@@ -19,7 +20,16 @@ class SelfKeyDecryption implements AtKeyDecryption {
           intent: Intent.decryptData,
           exceptionScenario: ExceptionScenario.decryptionFailed);
     }
-
+    if (atKey.key == "shared_key") {
+      if (atKey.sharedWith != _atClient.getCurrentAtSign()) {
```

Comment:
sharedWith is not set - when we create `shared_key.bob@alice`  - again, what happens when you run this branch with your REPL? Do you get the desired outcomes?